src/github.com/mozilla-services/pushgo/simplepush/dynamo_store.go

```diff
@@ -0,0 +1,507 @@
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				desc, err := table.DescribeTable()
+				if err != nil {
+					return
+				}
+				if desc.TableStatus == status {
+					done <- true
+					return
+				}
+				time.Sleep(s.statusIdle)
+			}
+		}
+	}()
+	select {
+	case <-done:
+		err = nil
+		break
+	case <-timeout:
+		err = ErrDynamoDBTimeout
+		close(done)
+	}
+	return
+}
+
+func (s *DynamoDBStore) createTable() (err error) {
+	/* There are three primary fields for each row,
+	   uaid = UserAgentID, all registered items must have one of these
+	   chid = Either the ChannelID or " ". In this case " " fills the
+	   		"main" UAID entry, which can carry info like the Proprietary
+	   		Ping data, etc. CHID rows can be accessed by specifing a
+	   		QUERY with chid > " ".
+	   created = UTC().Unix() of the record creation. Used for cleanup.
+	*/
+	tableDesc := dynamodb.TableDescriptionT{
+		TableName: s.tablename,
+		AttributeDefinitions: []dynamodb.AttributeDefinitionT{
+			dynamodb.AttributeDefinitionT{UAID_LABEL, "S"},
+			dynamodb.AttributeDefinitionT{CHID_LABEL, "S"},
+			dynamodb.AttributeDefinitionT{MODD_LABEL, "N"},
+		},
+		KeySchema: []dynamodb.KeySchemaT{
+			dynamodb.KeySchemaT{UAID_LABEL, "HASH"},
+			dynamodb.KeySchemaT{CHID_LABEL, "RANGE"},
+		},
+		GlobalSecondaryIndexes: []dynamodb.GlobalSecondaryIndexT{
+			dynamodb.GlobalSecondaryIndexT{
+				IndexName: "uaid-modified-index",
+				KeySchema: []dynamodb.KeySchemaT{
+					dynamodb.KeySchemaT{UAID_LABEL, "HASH"},
+					dynamodb.KeySchemaT{MODD_LABEL, "RANGE"},
+				},
+				Projection: dynamodb.ProjectionT{"ALL"},
+				ProvisionedThroughput: dynamodb.ProvisionedThroughputT{
+					ReadCapacityUnits:  s.readProv,
+					WriteCapacityUnits: s.writeProv,
+				},
+			},
+		},
+		ProvisionedThroughput: dynamodb.ProvisionedThroughputT{
+			ReadCapacityUnits:  s.readProv,
+			WriteCapacityUnits: s.writeProv,
+		},
+	}
+
+	pk, err := tableDesc.BuildPrimaryKey()
+	if err != nil {
+		s.logger.Panic(DB_LABEL, "Could not create primary key",
+			LogFields{"error": err.Error()})
+		return
+	}
+	s.table = s.server.NewTable(tableDesc.TableName, pk)
+	_, err = s.server.CreateTable(tableDesc)
+	if err != nil {
+		s.logger.Panic(DB_LABEL, "Could not create",
+			LogFields{"error": err.Error()})
+		return
+	}
+
+	err = s.waitUntilStatus(s.table, "ACTIVE")
+	if err != nil {
+		s.logger.Panic("dynamo", "Could not file table.",
+			LogFields{"error": err.Error()})
+		return
+	}
+	return
+}
+
+// NewEmcee creates an unconfigured memcached adapter.
+func NewDynamoDB() *DynamoDBStore {
+	s := &DynamoDBStore{
+		statusTimeout: 3 * time.Minute,
+		statusIdle:    5 * time.Second,
+	}
+	return s
+}
+
+func (*DynamoDBStore) ConfigStruct() interface{} {
+	return &DynamoDBConf{
+		MaxChannels: 200,
+		TableName:   "simplepush",
+		Region:      "us-west-1",
+		Access:      "MISSING_ACCESS_KEY",
+		Secret:      "MISSING_SECRET_KEY",
+		ReadProv:    1,
+		WriteProv:   1,
```

Comment:
Well, 1/1 is a lot less than the free tier of 25/25. I was using that as a config mostly because I was just doing simple function testing. I'll bump these values to the free provisioning layer.